src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -44,6 +55,46 @@ public function iOpenProductPage(ProductInterface $product): void
         $this->client->show($product->getCode());
     }
 
+    /**
+     * @When I browse products from taxon :taxon
+     */
+    public function iBrowseProductsFromTaxon(TaxonInterface $taxon): void
+    {
+        $this->client->index();
+        $this->client->addFilter('productTaxons.taxon.code', $taxon->getCode());
+        $this->client->filter();
+    }
+
+    /**
+     * @Then /^I should see the product price ("[^"]+")$/
+     */
+    public function iShouldSeeTheProductPrice(int $price): void
+    {
+        Assert::true(
+            $this->hasProductWithPriceInChannel(
+                [$this->responseChecker->getResponseContent($this->client->getLastResponse())],
+                $price,
+                $this->sharedStorage->get('channel'),
+            )
+        );
+    }
+
+    /**
+     * @Then /^I should see the (product "[^"]+") with price ("[^"]+")$/
+     */
+    public function iShouldSeeTheProductWithPrice(ProductInterface $product, int $price): void
+    {
+        Assert::true(
+            $this->hasProductWithPriceInChannel(
+                $this->responseChecker->getCollection($this->client->getLastResponse()),
+                $price,
+                $this->sharedStorage->get('channel'),
+                $product->getCode()
+            ),
+            sprintf("there is no product with %s code and %s price", $product->getCode(), $price)
```

Comment:
```suggestion
            sprintf("There is no product with %s code and %s price", $product->getCode(), $price)
```
and probably the price should be converted to a proper format